refinery/core/api.py

```diff
@@ -831,6 +843,54 @@ def get_study_assays(self, request, **kwargs):
             study_uuid=kwargs['study_uuid']
         )
 
+    def handle_incomplete_dataset(self, dataset):
+        """
+        Handle an incomplete DataSet instance when a detail view is
+        accessed.
+        :param dataset: a DataSet object
+        :return: a DataSet object
+        :raises: A TastyPie NotFound Exception if said DataSet
+        doesn't have the proper information associated yet.
+        """
+        if dataset.get_version_details() is not None:
+            return dataset
+
+        else:
+            logger.error(
+                self.make_incomplete_dataset_error_message(dataset.uuid)
+            )
+            raise NotFound(
+                self.make_incomplete_dataset_error_message(dataset.uuid)
+            )
+
+    def handle_incomplete_datasets(self, datasets):
+        """
+        Handle incomplete DataSet instances when a list view is accessed.
+        :param datasets: a list of DataSet objects
+        :return: a list of DataSet objects sans ones that don't have the
+        proper information associated yet.
+        """
+        complete_datasets = []
+
+        for dataset in datasets:
+            if dataset.get_version_details() is None:
```

Comment:
I thought that it was okay to split these two up since they really are providing different functions:

`handle_incomplete_dataset` happens during a detail view, takes a single dataset and returns a 404 if its not fully created.

`handle_incomplete_datasets` happens during a list view, takes a list of datasets and excludes datasets from its response if they are not fully created, logging the errors rather than providing a 404 for the entire response.